name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/TargobankPDFExtractor.java

```diff
@@ -58,413 +68,529 @@ public String getLabel()
+                    catch (ClassCastException e1)
                     {
-                        a1 = (AccountTransaction) transactions.get(1).getSubject();
-                        a2 = (AccountTransaction) transactions.get(0).getSubject();
+                        try
+                        {
+                            sellTransaction = (BuySellEntry) transactions.get(1).getSubject();
+                            taxTransaction = (AccountTransaction) transactions.get(0).getSubject();
+                        }
+                        catch (ClassCastException e2)
+                        {
+                            // do nothing because no tax transaction is present
+                        }
                     }
 
-                    // the dividend document might have a gross_value unit,
-                    // which needs to be copied over
-                    Optional<Unit> unitGross = a2.getUnit(Unit.Type.GROSS_VALUE);
-                    if (unitGross.isPresent())
+                    // Check if there is a sell transaction and a tax
+                    // transaction
+                    if (AccountTransaction.Type.TAXES.equals(taxTransaction.getType()) && PortfolioTransaction.Type.SELL
+                                    .equals(sellTransaction.getPortfolioTransaction().getType()))
                     {
-                        a1.addUnit(unitGross.get());
-                    }
-                    // set correct date stored in attribute
-                    a1.setDateTime((LocalDateTime) a1.getSecurity().getAttributes()
-                                    .get(new AttributeType(ATTRIBUTE_PAY_DATE)));
+                        // Subtract the taxes from the tax transaction from the
+                        // total amount
+                        sellTransaction.setMonetaryAmount(sellTransaction.getPortfolioTransaction().getMonetaryAmount()
+                                        .subtract(taxTransaction.getMonetaryAmount()));
+
+                        // Add taxes as tax unit
+                        sellTransaction.getPortfolioTransaction()
+                                        .addUnit(new Unit(Unit.Type.TAX, taxTransaction.getMonetaryAmount()));
+
+                        // Combine at sources file
+                        sellTransaction.setSource(sellTransaction.getSource() + "; " + taxTransaction.getSource());
 
-                    // combine document notes and mark transaction 2 to be
-                    // deleted
-                    a1.setSource(concat(a2.getSource(), a1.getSource()));
-                    a1.setNote(concat(a2.getNote(), a1.getNote()));
-                    a2.setNote(TO_BE_DELETED);
+                        // Combine at notes
+                        sellTransaction.setNote(concat(sellTransaction.getNote(), taxTransaction.getNote()));
+
+                        // Set note that the tax transaction will be deleted
+                        taxTransaction.setNote(TO_BE_DELETED);
```

Comment:
You also have access to the Item itself (e.g. 474). Therefore instead of using the note to make it as deleted and iterate over the items later, you could also say ```items.remove( ... )``` here